### DISCUSSION TITLE
Training results

### DISCUSSION THREAD

**Discussion by maolimin** [anthonyweidai/SvANet#31] [General]
100%|████████████████████████████████| 300/300 [01:34<00:00,  3.19it/s]
100%|████████████████████████████████| 100/100 [00:20<00:00,  4.90it/s]
Epoch: [35/100]         CrossValid: [1/1],      Learning rate: 0.000371
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.2035,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:35<00:00,  3.16it/s]
100%|████████████████████████████████| 100/100 [00:19<00:00,  5.07it/s]
Epoch: [36/100]         CrossValid: [1/1],      Learning rate: 0.000364
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1979,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:34<00:00,  3.19it/s]
100%|████████████████████████████████| 100/100 [00:19<00:00,  5.12it/s]
Epoch: [37/100]         CrossValid: [1/1],      Learning rate: 0.000357
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1925,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:33<00:00,  3.22it/s]
100%|████████████████████████████████| 100/100 [00:19<00:00,  5.10it/s]
Epoch: [38/100]         CrossValid: [1/1],      Learning rate: 0.000350
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1874,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:35<00:00,  3.15it/s]
100%|████████████████████████████████| 100/100 [00:20<00:00,  4.94it/s]
Epoch: [39/100]         CrossValid: [1/1],      Learning rate: 0.000342
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1826,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:33<00:00,  3.20it/s]
100%|████████████████████████████████| 100/100 [00:19<00:00,  5.23it/s]
Epoch: [40/100]         CrossValid: [1/1],      Learning rate: 0.000335
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1781,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:34<00:00,  3.18it/s]
100%|████████████████████████████████| 100/100 [00:20<00:00,  4.93it/s]
Epoch: [41/100]         CrossValid: [1/1],      Learning rate: 0.000328
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,
ValmDice: 100.0, Loss: 0.0, ValLoss: 0.0, ValmEM: 25.0, ValmAE: 0.1737,
AvgRecall: 100.0, AvgPrecis: 100.0, AvgF2Score: 100.0,
100%|████████████████████████████████| 300/300 [01:33<00:00,  3.21it/s]
100%|████████████████████████████████| 100/100 [00:19<00:00,  5.12it/s]
Epoch: [42/100]         CrossValid: [1/1],      Learning rate: 0.000320
Best: 100.0, BEpoch: 6, mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0,

My training command CUDA_VISIBLE_DEVICES=0,1,2,3 python main.py --task segmentation --gpus 3 --setname FIVES --num_repeat 3 --num_workers 2 --epochs 100 --save_point None --batch_size 4 --lr 1e-6 --warmup_init_lr 5e-5 --max_lr 5e-4 --model_name resnet50 --sup_method common --optim adamw --weight_decay 0.03 --seg_head_name deeplabv3 --seg_feature_guide 2 --fg_start_stage 1 --pretrained 2 --freeze_weight 3 --weight_name resnet50-11ad3fa6.pth   ,tell me  why? mIoU: 100.0, ValmIoU: 100.0, mDice: 100.0 ?

**Comment by anthonyweidai** [comment]
You may need to check with your dataset folder following https://github.com/anthonyweidai/SvANet/blob/main/readme/DATA.md.

**Comment by maolimin** [comment]
I noticed that in your code, for the images in the mask folder, the mask needs to have its foreground values changed from 255 to 1. I checked your project code, and in the segmentation parameter initialization it says:

opt.mask_fill = 255
opt.ignore_idx = 255,Did you convert the mask to 0–1

**Comment by anthonyweidai** [comment]
I didn't convert the multi-class mask to binary mask. If not, all foreground objects will be considered as the same object. If your data has over 255 classes, the value should be changed. This value means you don't need to calculate the pixel with value of 255 in the loss accumulation. You can see the loss function, there is a ignore index parameter. mask_fill is something that you fill in blank regions in data augmentation and the blank regions are different from the background region in original images.

**Comment by maolimin** [comment]
<img width="862" height="1108" alt="Image" src="https://github.com/user-attachments/assets/5649817e-5d52-4020-956c-3de52926964b" />“I saw what you told someone else, and now I’ve done the same operation on the FIVES data.”  And “I noticed that the validation loss was abnormally high in the first and second rounds.” 

<img width="865" height="929" alt="Image" src="https://github.com/user-attachments/assets/0487f22a-e127-45af-bcd8-d99d8c5ec2f3" />

**Comment by maolimin** [comment]
This code uses the test folder as the validation set during training, which is quite unusual.

I checked the code, and the workflow is like this:

Reading the data paths

In lib/utils/path_manage/utils.py, getImgPath(..., Mode=4) reads a directory structure like yours:

dataset/FIVES/
├─ train/
├─ test/
└─ mask/

as:

TrainSet = dataset/FIVES/train/*
TestSet = dataset/FIVES/test/*

In other words, the test set here is actually the split being used for validation during training.
